worlds/civ_6/Regions.py

```diff
@@ -0,0 +1,161 @@
+    if exclude_progressive_items:
+        flat_progressive_items = get_flat_progressive_districts()
+        prereqs_without_progressive_items: List[str] = []
+        for item in era_required_items:
+            if item in flat_progressive_items:
+                continue
+            prereqs_without_progressive_items.append(item)
+
+        return [get_item_by_civ_name(prereq, item_table) for prereq in prereqs_without_progressive_items]
+
+    return [get_item_by_civ_name(prereq, item_table) for prereq in era_required_items]
+
+
+def has_required_progressive_districts(state: CollectionState, era: EraType, player: int) -> bool:
+    """ If player has progressive items enabled, it will count how many progressive techs it should have, otherwise return the default array"""
+    progressive_districts = get_progressive_districts_data()
+
+    item_table = state.multiworld.worlds[player].item_table
+    # Verify we can still reach non progressive items
+    all_previous_items_no_progressives = get_prereqs_for_era(
+        era, True, item_table)
+    if not state.has_all([item.name for item in all_previous_items_no_progressives], player):
+        return False
+
+    # Verify we have the correct amount of progressive items
+    all_previous_items = get_prereqs_for_era(
+        era, False, item_table)
+    required_counts: Dict[str, int] = {}
+
+    for key, value in progressive_districts.items():
+        required_counts[key] = 0
+        for item in all_previous_items:
+            if item.civ_name in value:
+                required_counts[key] += 1
+
+    return state.has_all_counts({format_item_name(key): value for key, value in required_counts.items()}, player)
+
+
+def has_required_progressive_eras(state: CollectionState, era: EraType, player: int) -> bool:
+    """Checks, for the given era, how many are required to proceed to the next era. Ancient = 0, Classical = 1, etc."""
+    if era == EraType.ERA_FUTURE or era == EraType.ERA_INFORMATION:
+        return True
+
+    eras = [e.value for e in EraType]
+    era_index = eras.index(era.value)
+    return state.has(format_item_name("PROGRESSIVE_ERA"), player, era_index + 1)
+
+
+def has_required_items(state: CollectionState, era: EraType, world: 'CivVIWorld') -> bool:
+    player = world.player
+    has_progressive_districts = world.options.progression_style != "none"
+    has_progressive_eras = world.options.progression_style == "eras_and_districts"
+
+    if has_progressive_districts:
+        required_items = has_required_progressive_districts(state, era, player)
+    else:
+        era_required_items = [get_item_by_civ_name(item, world.item_table).name for item in get_era_required_items_data()[era.value]]
+        required_items = state.has_all(era_required_items, player)
+
+    if not required_items:
+        return False
+
+    return not has_progressive_eras or has_required_progressive_eras(state, era, player)

```

Comment:
You could do this, though it does get a little complicated, up to you
```suggestion
    return required_items and (not has_progressive_eras or has_required_progressive_eras(state, era, player))
```

Reply:
Nice, I like it